pkg/reconciler/common/releases.go

```diff
@@ -23,17 +23,30 @@ import (
 	"path"
 	"path/filepath"
 	"sort"
+	"strings"
 
 	mf "github.com/manifestival/manifestival"
 	"golang.org/x/mod/semver"
 	"knative.dev/operator/pkg/apis/operator/v1alpha1"
 )
 
 const (
+	// KoEnvKey is the key of the environment variable to specify the path to the ko data directory
 	KoEnvKey = "KO_DATA_PATH"
+
+	// URLLinkTemplate is the template of the URL used to access the released artifacts of Knative
+	URLLinkTemplate = "https://github.com/knative/%s/releases/download/v%s/%s"
 )
 
-var cache = map[string]mf.Manifest{}
+var (
+	cache = map[string]mf.Manifest{}
+
+	// ServingYamlNames is the string array containing all the artifacts to download for Knative serving
+	ServingYamlNames = []string{"serving-crds.yaml", "serving-core.yaml", "serving-hpa.yaml", "serving-storage-version-migration.yaml"}
+
+	// EventingYamlNames is the string array containing all the artifacts to download for Knative eventing
+	EventingYamlNames = []string{"eventing.yaml", "upgrade-to-%s.yaml", "storage-version-migration-%s.yaml"}
```

Comment:
I don't think we can assume these files will be a part of every version we might fetch. I think we need to determine these dynamically instead of hard-coding them.

Reply:
@jcrossley3 How can we dynamically know them?

Reply:
I don't know. It may require some upstream operand coordination. But if the answer is "we can't" then I don't think we can claim to be able to reliably install a future operand release without cutting a corresponding operator release.

Reply:
@jcrossley3 The path is based on the template `https://github.com/knative/%s/releases/download/v%s/%s`. The first %s is the component and the second is version number.
EventingYamlNames and ServingYamlNames save the file names, which are consistent for all releases(at least for now). As long as Knative still release the same files, we are fine to read them dynamically for each release.

Reply:
So when say, "serving-storage-version-migration.yaml", is removed from a release, what then? Cut a new release of the operator? You'll still have to deal with the fact that some versions supported by the operator contain that file and some don't.

Reply:
I agree I think we need a naming scheme that we agree with upstream.

I'd expect `serving-crds` and `serving-core` to be stable. The jobs might need a prefix or something so we can list the artifacts through the releases API and fetch all of the upgrade jobs necessary. Or we agree with the upstream's that there's always a YAML called `serving-upgrade.yaml`, that contains all necessary upgrade resources.

I agree we can't hardcode those as is.

Reply:
https://github.com/knative/serving/issues/8436
https://github.com/knative/eventing/issues/3386
I logged two issues for serving and eventing, to track this request.